tests/MongoDB.Driver.Core.TestHelpers/CoreTestConfiguration.cs

```diff
@@ -107,14 +108,17 @@ public static ClusterBuilder ConfigureCluster(ClusterBuilder builder)
                 .ConfigureWithConnectionString(__connectionString.Value)
                 .ConfigureCluster(c => c.With(serverSelectionTimeout: TimeSpan.FromMilliseconds(int.Parse(serverSelectionTimeoutString))));
 
-            if (__connectionString.Value.Tls.HasValue && __connectionString.Value.Tls.Value)
+            if (__connectionString.Value.Tls.HasValue &&
+                __connectionString.Value.Tls.Value &&
+                __connectionString.Value.AuthMechanism != null &&
+                __connectionString.Value.AuthMechanism == MongoDBX509Authenticator.MechanismName)
```

Comment:
Our AUTH tests currently default to SCRAM-SHA authentication, but a patch build could easily be modified to use X509 auth by specifying the proper auth mechanism